src/shared/json.c

```diff
@@ -4993,6 +4995,60 @@ int json_dispatch_unbase64_iovec(const char *name, JsonVariant *variant, JsonDis
         return 0;
 }
 
+int json_dispatch_byte_array_iovec(const char *name, JsonVariant *variant, JsonDispatchFlags flags, void *userdata) {
+        _cleanup_free_ uint8_t *buffer = NULL;
+        struct iovec *iov = ASSERT_PTR(userdata);
+        size_t sz, k = 0;
+
+        assert(variant);
+
+        if (!json_variant_is_array(variant))
+                return json_log(variant, flags, SYNTHETIC_ERRNO(EINVAL), "JSON field '%s' is not an array.", strna(name));
+
+        sz = json_variant_elements(variant);
+
+        buffer = new(uint8_t, sz);
```

Comment:
doesn't matter, but for other cases like this we pretty systematically append an extra NUL byte just for safety reasons to the end of decoded buffers. see unbase64mem() for example. i.e. maybe just replace sz by sz+1 here, and end the buffer in a NUL byte, simply for paranoia.

Reply:
Yeah, I considered that, but this is for binary data...

Reply:
yeah, same as unbase64mem() actually. but like unbase64mem() this is super generic code, and for thta kind of stuff i'd always try to be defensive.

but doesn't matter

Reply:
-> #30893.